backend/src/modules/doubles-tournament/core/application/command/CallMatchCommandHandler.ts

```diff
@@ -0,0 +1,37 @@
+import { CommandHandler } from '../../../../../shared/core/application/command/CommandHandler';
+import { CallMatch } from './CallMatch';
+import { DomainEventPublisher } from '../../../../../shared/core/application/event/DomainEventBus';
+import { CommandResult } from '../../../../../shared/core/application/command/CommandResult';
+import { MatchWasCalled } from '../../domain/event/MatchWasCalled';
+import { CurrentTimeProvider } from '../../../../../shared/core/CurrentTimeProvider';
+import { CommandPublisher } from '../../../../../shared/core/application/command/CommandBus';
+import { AssignTournamentTables } from '../../../../tournament-tables/core/application/command/AssignTournamentTables';
+
+export class CallMatchCommandHandler implements CommandHandler<CallMatch> {
+  constructor(
+    private readonly eventPublisher: DomainEventPublisher,
+    private readonly currentTimeProvider: CurrentTimeProvider,
+    private readonly commandPublisher: CommandPublisher,
+  ) {}
+
+  async execute(command: CallMatch): Promise<CommandResult> {
+    const matchWasCalled = new MatchWasCalled({
+      occurredAt: this.currentTimeProvider(),
+      tournamentId: command.tournamentId,
+      matchFromQueue: command.matchFromQueue,
+      table: command.table,
+    });
+
+    const tableNumber = command.table.tableNumber;
+    const tableName = command.table.tableName;
+    const availableToPlay: boolean = command.table.availableToPlay;
+    const tableToAssign: { tableNumber: number; tableName: string; availableToPlay?: boolean }[] = [
+      { tableNumber, tableName, availableToPlay },
+    ];
+
+    await this.commandPublisher.execute(new AssignTournamentTables(command.tournamentId, tableToAssign));
```

Comment:
@nowakprojects tak ma wyglądać wywołanie komendy z innego modułu za pomocą commandHandlera? 
execute w execute?

Reply:
Jeśli będzie odpowiedni command to wygląda raczej OK :) 
Zastanawia mnie jak odbierzesz potem po skonczonym meczu, że mecz skonczyl się mecz nr jakiś na turnieju o jakimś id? Nie trzeba tutaj czegos zapisac? Albo jakos inaczej sprytnie zrobić?

Albo dać jakąś ograniczoną liczbę wywołań? Ale zacząłbym od max jednego. Zazwyczaj na turnieju jest, że chyba po 3 (jesli dobrze pamietam @lamparina) jest walkower. To byloby w sumie ciekawe do implementacji... jesli zdarza sie 3 eventy MatchWasCalled dla jednego meczu to zapisz wynik walkoweru :) Ale tego nie trzeba teraz robic - moze byc inny task, a fajnie rozszerzalny mamy nasz system.

Reply:
"Zastanawia mnie jak odbierzesz potem po skonczonym meczu, że mecz skonczyl się mecz nr jakiś na turnieju o jakimś id? Nie trzeba tutaj czegos zapisac? Albo jakos inaczej sprytnie zrobić?"

ale to już w tasku obok jest, więc jak dojdę do niego to wtedy najwyżej wrócę tutaj, bo tak do przodu w sumie nie wybiegłem jeszcze :-)


Oho to tutaj zaczyna robić się ciekawie :-P